histoqc/tests/test_images_tsv_results.py

```diff
@@ -0,0 +1,152 @@
+
+    def setUp(self):
+
+        # image suffixes
+        self.suffixes = [
+            'areathresh',
+            'blurry',
+            'bright',
+            'dark',
+            'deconv_c0',
+            'deconv_c1',
+            'deconv_c2',
+            'equalized_thumb',
+            'fatlike',
+            'flat',
+            'fuse',
+            'hist',
+            'mask_use',
+            'small_fill',
+            'small_remove',
+            'spur',
+            'thumb_small',
+            'thumb'
+        ]
+
+        # tsv 
+        self.rs_name = "results.tsv"
+        # tsv_labels #### '#start_time:', '#outdir:', '#command_line_args:'
+        self.tsv_labels = ['#pipeline:',  '#config_file:', '#dataset:']
+        # tsv dataset fields
+        self.tsv_dataset_fields = ['filename','comments','image_bounding_box','base_mag','type','levels','height','width','mpp_x','mpp_y','comment','brightestPixels','dark','flat_areas','fatlike_tissue_removed_num_regions','fatlike_tissue_removed_mean_area','fatlike_tissue_removed_max_area','fatlike_tissue_removed_percent','small_tissue_filled_num_regions','small_tissue_filled_mean_area','small_tissue_filled_max_area','small_tissue_filled_percent','small_tissue_removed_num_regions','small_tissue_removed_mean_area','small_tissue_removed_max_area','small_tissue_removed_percent','background_contrast','background_contrast_std','background_dissimilarity','background_dissimilarity_std','background_homogeneity','background_homogeneity_std','background_ASM','background_ASM_std','background_energy','background_energy_std','background_correlation','background_correlation_std','background_tenenGrad_contrast','background_michelson_contrast','background_rms_contrast','background_grayscale_brightness','background_grayscale_brightness_std','background_chan1_brightness','background_chan1_brightness_std','background_chan2_brightness','background_chan2_brightness_std','background_chan3_brightness','background_chan3_brightness_std','blurry_removed_num_regions','blurry_removed_mean_area','blurry_removed_max_area','blurry_removed_percent','spur_pixels','areaThresh','template1_MSE_hist','template2_MSE_hist','template3_MSE_hist','template4_MSE_hist','final_contrast','final_contrast_std','final_dissimilarity','final_dissimilarity_std','final_homogeneity','final_homogeneity_std','final_ASM','final_ASM_std','final_energy','final_energy_std','final_correlation','final_correlation_std','tenenGrad_contrast','michelson_contrast','rms_contrast','grayscale_brightness','grayscale_brightness_std','chan1_brightness','chan1_brightness_std','chan2_brightness','chan2_brightness_std','chan3_brightness','chan3_brightness_std','chan1_brightness_YUV','chan1_brightness_std_YUV','chan2_brightness_YUV','chan2_brightness_std_YUV','chan3_brightness_YUV','chan3_brightness_std_YUV','deconv_c0_mean','deconv_c0_std','deconv_c1_mean','deconv_c1_std','deconv_c2_mean','deconv_c2_std','pixels_to_use','warnings']
```

Comment:
'#start_time:', '#outdir:', '#command_line_args:' fields will or could change each time.

Reply:
I was referring to the **tsv_dataset_fields**, and the **image suffixes**. Those will evolve as HistoQC is developed, and should obtained programmatically so that we do not have to keep updating the lists.

I think the tsv labels can be hard-coded. Can you help me understand what would cause the tsv_labels to change between runs?